appinventor/blocklyeditor/src/blocks/control.js

```diff
@@ -416,6 +416,140 @@ Blockly.Blocks['controls_forEach'] = {
   typeblock: [{translatedName: Blockly.Msg.LANG_CONTROLS_FOREACH_INPUT_ITEM}]
 };
 
+Blockly.Blocks['controls_for_each_dict'] = {
+  category: 'Control',
+
+  helpUrl: Blockly.Msg.LANG_CONTROLS_FOREACH_DICT_HELPURL,
+
+  init: function() {
+    console.log('init');
+    this.setColour(Blockly.CONTROL_CATEGORY_HUE);
+    this.appendValueInput('DICT')
```

Comment:
While this is technically correct (the best kind of correct), I'd like us to move more toward using `interpolateMsg` as a means of constructing blocks since I think it helps reduce the number of strings that need to be internationalized (we're at over 30k strings so it's a bit daunting). Please rewrite the block definition to use that approach.

Reply:
Q: I've seen other organizations (ex: [scratch](https://github.com/LLK/scratch-blocks/blob/b7088bb4046c3ffa4400d381238b260bce16d12e/blocks_vertical/motion.js#L37)) pass the block "shape" as json, but wrap inside a JS block definition for easy extension. I think it's a cool idea, but I'm not sure if there are downsides/what they may be. Is that something you'd be interested in?

Reply:
I'm not really sure that it is a level of indirection that we need at this time. It's also the case that because of things like lexical scoping of variables, etc. App Inventor blocks have more complex semantics than we might be able to capture in a JSON structure (I'd need to think about that more), so while this may work for most blocks it may not work for all, and it's not clear that a complete rewrite to the JSON form would bring much benefit.